### ISSUE TITLE
[Survival Beta] Suggestion - Prevent player from salvaging blocks from wrecks (and add progression and component salvage)

### ISSUE THREAD

**Issue by DrizztCode** [tsunamayo/Starship-EVO#5629]
Hi

my first suggestion is as the title says - since the player is able to salvage all the different block types from the wrecks as soon as they craft a placeable block - which not only prevents the need for the player to build anything other than a means to fuel their ship - it also gives them lightcruise engines - as well as there possibly being some odd effects when using salvaged blocks (but that one is not confirmed, since other factors would have been in play)

my suggestion would be for wrecks to have a flag that prevents the player from using the right click to deconstruct them

i would then have a progression mechanism for the mining laser, where at first the player can only salvage scrap
and then after they have salvaged x number of pieces of scrap (maybe 1000?), then will then get some components when they mine blocks (component type and yield depending on the block type and player skill) - which higher levels increasing the yield and the types of components that can be gained - possibly maybe have a "perk" type system where the player can choose to increase component yield or component level for each skill increase

and higher levels will be dependent on salvaging higher number of more advanced components

i would also suggest that the first player wreck is more wrecked - or maybe a small scout ship - maybe only big enough that they could get to the first progression level by scrapping the whole wreck - and would then have very few pieces left to salvage - so that they can't get too many components that way too early - but there will then be a lot of incentive to find more wrecks

i would also suggest having the wrecks spawn much further apart

i think there is also an argument for having more advanced block types locked behind this progression system - maybe requiring the player to scrap the block type when at high enough level

examples would be the lightcruise engine and warp drive - so the player would not only have to get crystals from the asteroids to build lightcruise drives - they would also have to spend some time levelling up and exploring the planet - and then for warp drives they could need both the warpitron crystals (yes that is a lame suggestion for a name lol) - and also to find a warp drive to salvage - maybe on a wreck on the planet with the crystals - or maybe more fun would be floating wrecks or from pirates etc

possibly also other ways to increase tech skills - maybe more appropriate achievement builder-y based things - like building a ship of a certain size or with a specific set of capabilities

but maybe those challenges can unlock other things and the salvage tech tree is all about salvaging x number of components and salving the block type with a high enough skill level

anyway - i hope this all makes sense and seems like a good idea - i will of course be delighted to hear what people think and to discuss the options :-)

Thanks for all the great work Tsuna! :-)

**Comment by ProPeach** [comment]
Perhaps there could be levels of shipwreck, with the starter planet only featuring "low level" wrecks which don't have advanced systems - no Lightcruise thrusters etc. Wrecks at the moment are the only way we can get medbays which are pretty fundamental to early progression, although I'm sure we'll be able to craft those soon

**Comment by DrizztCode** [comment]
> Perhaps there could be levels of shipwreck, with the starter planet only featuring "low level" wrecks which don't have advanced systems - no Lightcruise thrusters etc. Wrecks at the moment are the only way we can get medbays which are pretty fundamental to early progression, although I'm sure we'll be able to craft those soon

i agree about the medbay (although i didn't actually salvage one and had to go the the space station when i almost suffocated in the habitat lol) - although yes i also think they are one of several parts not yet in the assembler directory (as well as hoses - i love hoses - i miss hoses :-( )

but i would say that apart from that specific case of the medbay - there is no good reason to allow early game salvage of entire blocks - maybe in mid or late game the player would have the ability to do it - either by levelling up their salvage skill, or by acquiring a piece of equipment that enables it

i do like the idea of being able to salvage components though, but with the progression i mentioned - so for the first level up you can salvage metal plates and maybe some other tier one components, in low quantities - then next level you can choose to get a higher yield of those or add frames or some other higher tier component - and each level up could require a certain number of salvage points - with higher tier components having more points - but if the player selects to have a higher yield of lower tier components they still get an equivalent rate of levelling due to getting e.g. 20 plates at 1 point each, instead of 4 frames at 5 points each for the same amount of time spent salvaging

or maybe there would be a small levelling penalty for choosing the higher yield, but the player may decide that the higher rate of those particular components is more valuable to them at that time than levelling faster

definitely lots to think about :-)

and now i am going to return to the first wreck in my new game and see if i can find the medbay lol ;-)

**Comment by DrizztCode** [comment]
i would also add that once there is limited inventory, the player salvaging entire blocks by hand would also be somewhat moot. Maybe instead a ship mounted piece if equipment could do it - maybe the players links a hand tool to the ship equipment and any salvaged blocks flow into the ship inventory through a cargo scoop.

plus i am about to make a suggestion that blocks get built using components from the inventory - which will also make it moot too (with full block salvaging just giving all of the components for a block instead of just a few)

**Comment by tsunamayo** [comment]
@DrizztCode Are you still able to delete and get back block from wrecks? This should have been fixed and I cant make it work.

**Comment by DrizztCode** [comment]
> @DrizztCode Are you still able to delete and get back block from wrecks? This should have been fixed and I cant make it work.

i am still able to salvage entire blocks from wrecks by right clicking on 24w09c - i did think i couldn't, because for some reason it did not work for the first game i started on this build - but when i tried later it did work - i have had a few occasions where the red x shows up and it does not allow it - but mostly it lets me reclaim any block from a wreck

**Comment by DrizztCode** [comment]
> > @DrizztCode Are you still able to delete and get back block from wrecks? This should have been fixed and I cant make it work.
> 
> hi - i am still able to salvage entire blocks from wrecks by right clicking (on 24w09c) - i did think i couldn't, because for some reason it did not work for the first game i started on this build - but when i tried later it did work - i have had a few occasions where the red x shows up and it does not allow it - but mostly it lets me reclaim any block from a wreck

however.................i just reloaded the new game i started to test this, which let me remove parts (with some scrappy hull in hand)

and now it won't let me

i will do some more testing

**Comment by DrizztCode** [comment]
> > > @DrizztCode Are you still able to delete and get back block from wrecks? This should have been fixed and I cant make it work.
> > 
> > 
> > hi - i am still able to salvage entire blocks from wrecks by right clicking (on 24w09c) - i did think i couldn't, because for some reason it did not work for the first game i started on this build - but when i tried later it did work - i have had a few occasions where the red x shows up and it does not allow it - but mostly it lets me reclaim any block from a wreck
> 
> however.................i just reloaded the new game i started to test this, which initially let me remove parts (with some scrappy hull in hand)
> 
> and now it won't let me
> 
> i will do some more testing

so i then went searching for another wreck and it did let me salvage blocks

so i then exited and reloaded - and it still let me

so i think that it is only very rare that it does not allow blocks to be salvaged

**Comment by tsunamayo** [comment]
@DrizztCode Survival is finally released, with a lot of stuff you were suggesting! Took a long time, I hope you get to enjoy the game.
